### PR TITLE
Atm and test atm update

### PR DESCRIPTION
**Pull request by Fikre-Mamo**
Refactor ATM system code: 

- Added shebang and encoding declaration to the top of the main script and test file.
- Renamed test file to `test_atm.py` to follow naming conventions.
- Added module-level docstrings to both the main script and the test file.
- Replaced `print` statements with return values in ATM functions for better separation of concerns.
- Updated `show_balance` function to return messages instead of printing it.

### CODE REVIEW

File: solutions/tests/tast_atm.py [hunk 1]
```diff
@@ -1,11 +1,16 @@
...
 Created on: 2025/01/02
 By: Fikremichael Mamo
 """
 
 import unittest
-from atm import show_balance, deposit, withdraw
+from ..atm import show_balance, deposit, withdraw
 
 
 class TestATM(unittest.TestCase):
```
Comment (sara-shahin-cell) [inline solutions/tests/tast_atm.py]: the test class should have a docstring 
ex: """ unit test for atm machine function"""

File: solutions/atm.py [hunk 2]
```diff
@@ -15,7 +20,7 @@ def show_balance(balance):
     Parameters:
         balance (float): The current balance in the user's account.
```
Comment (Nouranaia) [inline solutions/atm.py:21]: there is no description of the amount parameter, please add (The function's arguments are described:
  - [ ] Type
  - [ ] Purpose)

Comment (sara-shahin-cell) [inline solutions/atm.py:19]: the other assumptions are not documented please add documentation

File: solutions/atm.py [hunk 3]
```diff
@@ -15,7 +20,7 @@ def show_balance(balance):
...
     """
-    print(f"Your Balance is ${balance:.2f}")
```
Comment (sara-shahin-cell) [inline solutions/atm.py:18]: please don't use `print` statements anywhere

File: solutions/atm.py [hunk 4]
```diff
@@ -15,7 +20,7 @@ def show_balance(balance):
...
+    return f"Your Balance is ${balance:.2f}"
```
Comment (sara-shahin-cell) [inline solutions/atm.py:24]: - please add defensive assertions
ex: 
assert isinstance (balance, (float, int)), "Input must be a number (float)."

**Review by sara-shahin-cell — CHANGES_REQUESTED** [review]
# Behavior

## Files

- [x] The file name describes the function's behavior
- [x] There is a module docstring in the function file
- [x] The test file's name matches the function file name -
  `/tests/test_file_name.py`
- [x] There is a module docstring in the tests file

### Unit Tests

- [x] The test class has a helpful name in PascalCase
- [ ] _The test class has a docstring_ **please  add docstring to test class**
- _Every unit test has_ **no unit test exists. please add unit tests**
  - [x] A helpful name
  - [x] A clear docstring
  - [x] Only one assertion
  - [x] There is no logic in the unit test
- [x] All tests pass
- [x] There are tests for defensive assertions
- [x] There are tests for boundary cases

### Function Docstring

- [x] The function's behavior is described
- The function's arguments are described: 
  - [x] Type
  - [x] Purpose
  - [ ] Other assumptions (eg. if it's a number, what's the expected range?) 
- The return value is described
  - [x] Type
  - [ ] Other assumptions are documented **it's not documented please add documentation**
- The defensive assertions are documented using `Raises:` **no documentation for defensive assertions please add**
  - [ ] Each assumption about an argument is checked with an assertion
  - [ ] Each assertion checks for _only one_ assumption about the argument
- [ ] Include 3 or more (passing!) doctests **there is no  doctests please add**

### The Function

- [x] The function's name describes it's behavior
- [x] The function's name matches the file name
- [x] The function has correct type annotations
- [x] The function is not called in the function file

## Strategy

### Do's

- [x] Variable names help to understand the strategy
- [x] Any comments are clear and describe the strategy
- [x] Lines of code are spaced to help show different stages of the strategy

### Don'ts

- [x] The function's strategy _is not_ described in the documentation
- [x] Comments explain the _strategy_, **not** the _implementation_
- [x] The function _does not_ have more comments than code
  - If it does, consider finding a new strategy or a simpler implementation

## Implementation

- [x] The code passes the formatting checks
- [x] The code passes all Ruff linting checks
- [x] The code has no (reasonable) Pylint errors
  - In code review, you can decide when fixing a Pylint error is helpful and
    when it's too restricting.
- [x] Variables are named with snake_case
- [x] Variable names are clear and helpful
- [x] The code follows the strategy as simply as possible
- [x] The implementation is as simple as possible given the strategy
- [x] There are no commented lines of code
- [ ] There are no `print` statements anywhere **please don't use `print` statements anywhere**
- [ ] The code includes defensive assertions **please add defensive assertions**
- [ ] Defensive assertions include as little logic as possible
